app/templates/teachers/teacher-student-view.jade

```diff
@@ -0,0 +1,220 @@
+        //-                   //- - var levelAverage = view.allLevelSessionMap[level.get('original')]
+        //-                   p= levelNumber + ". " + level.get('name') + ": " + session.get('playtime') + " (average: " + ")"
+        //-               p= coursePlaytime 
+
+        div.student-levels 
+          h4(data-i18n="teacher.course_progress")
+          div.student-levels-progress
+            each versionedCourse in view.classroom.get('courses') || []
+                - var course = _.find(view.courses.models, function(c) {return c.id === versionedCourse._id;});
+                if !course
+                  - continue;
+                - var instance = view.courseInstances.findWhere({ courseID: course.id, classroomID: view.classroom.id })
+                - if (instance && instance.hasMember(view.user))
+                  if course
+                    div.course-row.alternating-background
+                      div.course-info= course.get('name')
+                      - var levels = view.classroom.getLevels({courseID: course.id}).models
+                      each level, index in levels
+                        - var levelNumber = view.classroom.getLevelNumber(level.get('original'), index + 1)
+                        //- - console.log (level)
+                        - var levelProgress = view.levelProgressMap[level.get('original')]
+                        //- p= levelNumber + ". " + level.name + " Progress: " + view.levelProgressMap[level.original]
+                        //- - console.log(level)
+                        +studentLevelProgressDot(levelProgress, level, levelNumber, course)
+
+      
+                      
+        if status == 'enrolled'
+          div
+            h4
+              span= view.user.get('name')
+              span.spr
+              span(data-i18n="teacher.not_assigned")
+              span :
+            each versionedCourse in view.classroom.get('courses') || []
+              - var course = _.find(view.courses.models, function(c) {return c.id === versionedCourse._id;});
+              if !course
+                - continue;
+              - var instance = view.courseInstances.findWhere({ courseID: course.id, classroomID: view.classroom.id })
+              - if (!(instance && instance.hasMember(view.user)))
+                  div(class="small-details")
+                    span=course.get('name')
+                  // add a button to assign this course 
+                
+        //- else
+        //-   p Apply a license to this student to assign additional courses!
+        //-   button.enroll-student-button.btn.btn-navy(data-i18n="teacher.apply_license", data-user-id=view.user.id, data-event-action="Teachers Class Enrollment Enroll Student")
+        //-   // this button doesn't work yet
+        
+
+mixin studentLevelProgressDot(levelProgress, level, levelNumber, course)
+  //- TODO: Refactor with TeacherClassesView jade
```

Comment:
Code is not entirely identical, since it uses a slightly different datasource. What would you recommend I do for now?